custom_components/google_home/config_flow.py

```diff
@@ -51,22 +51,27 @@ async def async_step_user(
 
         if user_input is not None:
             username = user_input[CONF_USERNAME]
+            password = user_input[CONF_PASSWORD]
             session = async_create_clientsession(self.hass)
-            client = GlocaltokensApiClient(
-                hass=self.hass,
-                session=session,
-                username=user_input[CONF_USERNAME],
-                password=user_input[CONF_PASSWORD],
-            )
-            master_token = await self._test_credentials(client)
-            if master_token is not None:
-                config_data: dict[str, str] = {}
-                config_data[CONF_USERNAME] = user_input[CONF_USERNAME]
-                config_data[CONF_PASSWORD] = user_input[CONF_PASSWORD]
-                config_data[CONF_MASTER_TOKEN] = master_token
-                config_data[CONF_ANDROID_ID] = await client.get_android_id()
-                return self.async_create_entry(title=username, data=config_data)
-            self._errors["base"] = "auth"
+
+            if len(password) >= 100:
+                self._errors["base"] = "pass-len"
+            else:
```

Comment:
Just a matter of coding preference but it's nice to have a first block of `if` being "the longest". 
Like:
```python
from .const import MAX_PASSWORD_LENGTH
...
if len(password) < MAX_PASSWORD_LENGTH:
  ... # all the major logic
else:
  self._errors["base"] = "pass-len"
```
but it's purely up to the way you like it :) 
However, please move out the static `100` into a `const` for easier re-usability and style :)